core/src/test/java/org/elasticsearch/indices/IndicesRequestCacheIT.java

```diff
@@ -354,4 +354,67 @@ public void testQueryRewriteDatesWithNow() throws Exception {
                 equalTo(0L));
     }
 
+    public void testCanCache() throws Exception {
+        assertAcked(client().admin().indices().prepareCreate("index").addMapping("type", "s", "type=date")
+                .setSettings(IndicesRequestCache.INDEX_CACHE_REQUEST_ENABLED_SETTING.getKey(), true, IndexMetaData.SETTING_NUMBER_OF_SHARDS,
+                        5, IndexMetaData.SETTING_NUMBER_OF_REPLICAS, 0)
+                .get());
+        indexRandom(true, client().prepareIndex("index", "type", "1").setRouting("1").setSource("s", "2016-03-19"),
+                client().prepareIndex("index", "type", "2").setRouting("1").setSource("s", "2016-03-20"),
+                client().prepareIndex("index", "type", "3").setRouting("1").setSource("s", "2016-03-21"),
+                client().prepareIndex("index", "type", "4").setRouting("2").setSource("s", "2016-03-22"),
+                client().prepareIndex("index", "type", "5").setRouting("2").setSource("s", "2016-03-23"),
+                client().prepareIndex("index", "type", "6").setRouting("2").setSource("s", "2016-03-24"),
+                client().prepareIndex("index", "type", "7").setRouting("3").setSource("s", "2016-03-25"),
+                client().prepareIndex("index", "type", "8").setRouting("3").setSource("s", "2016-03-26"),
+                client().prepareIndex("index", "type", "9").setRouting("3").setSource("s", "2016-03-27"));
+        ensureSearchable("index");
```

Comment:
The method name implies yellow though. I bet there are places where we don't *need* green.